.github/workflows/ci-windows-artifacts.yml

```diff
@@ -0,0 +1,83 @@
+# This workflow generates artifacts such as avifenc.exe and avifdec.exe for convenience.
+
+name: CI Windows Artifacts
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  build-static:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest]
+
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - name: Setup Visual Studio shell
+      if: runner.os == 'Windows'
+      uses: egor-tensin/vs-shell@9a932a62d05192eae18ca370155cf877eecc2202 # v2.1
+    - name: Cache external dependencies
+      id: cache-ext
+      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      with:
+        path: ext
+        key: ${{ runner.os }}-${{ hashFiles('ext/*.cmd') }}
+    - name: Print cmake version
+      run: cmake --version
+    - uses: ilammy/setup-nasm@13cbeb366c45c4379d3478cdcbadd8295feb5028 # v1.5.1
+      with:
+        version: 2.15.05
+    - uses: seanmiddleditch/gha-setup-ninja@8b297075da4cd2a5f1fd21fe011b499edf06e9d2 # v4
+    - run: pip install meson
+    - name: Build aom
+      if: steps.cache-ext.outputs.cache-hit != 'true'
+      working-directory: ./ext
+      run: ./aom.cmd
+    - name: Build dav1d
+      if: steps.cache-ext.outputs.cache-hit != 'true'
+      working-directory: ./ext
+      run: ./dav1d.cmd
+    - name: Build libyuv
+      if: steps.cache-ext.outputs.cache-hit != 'true'
+      working-directory: ./ext
+      run: ./libyuv.cmd
```

Comment:
Thank you for adding this CI workflow.

We should use clang-cl to build libyuv. The assembly code in libyuv is written in the GCC inline assembly format. Visual C++ doesn't support GCC inline assembly.

In appveyor.yml this is done as follows:

```
 - set "CC=clang-cl" && set "CXX=clang-cl"
 - libyuv.cmd
 - set "CC=" && set "CXX="
```

Reply:
I was wondering why these lines were there. Thanks for the explanation.

Done in https://github.com/AOMediaCodec/libavif/pull/1930.